Automattic/pocket-casts-android
# Tracks: Add Firebase opt out

| 📘 Project: #261  |
|:---:|

Adds check for Firebase opt out. 

(Also renames `AnalyticsHelper` to `FirebaseAnalyticsTracker` and moves it to the `analytics` module.)

Corresponding iOS PR: https://github.com/Automattic/pocket-casts-ios/pull/333

## To test

1. Launch the app
2. Tap around the app
3. Verify you see `Analytic event` logs
4. Go to Profile > Cog > Privacy
5. Enable the opt out
6. Tap around
7. Verify you don't see `Analytic event` logs

# Checklist
N/A
- Should this change be included in the release notes? If yes, please add a line in CHANGELOG.md
- Have you tested in landscape?
- Have you tested accessibility with TalkBack?
- Have you tested in different themes?
- Does the change work with a large display font?
- Are all the strings localized?
- Could you have written any new tests?
- Did you include Compose previews with any components?
